### PR TITLE
Merge-same-source-notes

### PR DESCRIPTION
**Pull request by ransurf**
:sparkles: add merge command
Merge notes of the same source #20

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -309,20 +348,36 @@ export default class FleetingNotesPlugin extends Plugin {
...
+			const linkText = this.app.metadataCache.fileToLinktext(
+				note.file,
+				sourcePath
+			);
+			unprocessedNoteString += template.replace("${linkText}", linkText);
+		});
+		return unprocessedNoteString;
+	}
+	//![[linkText]]
+	embedNotesToString(notes: Array<ObsidianNote>, sourcePath: string) {
```
Comment (matthewwong525) [inline src/main.ts:367]: This is essentially the same as `unprocessedNotesToString`. I'd say to get rid of `unprocessedNotesToString` and have this function include an optional "prefix" parameter that adds `- [ ]` when calling from unprocessed notes function

File: src/main.ts [hunk 2]
```diff
@@ -374,6 +430,40 @@ export default class FleetingNotesPlugin extends Plugin {
...
+							if (fm === source) {
+								hasSource = true;
+							}
+						}
+				  )
+				: null;
+			return hasSource;
+		};
+
+		const hasSourceInContent = (note: ObsidianNote) => {
```
Comment (matthewwong525) [inline src/main.ts:457]: smart to include this! didn't think of this

File: src/main.ts [hunk 3]
```diff
@@ -374,6 +430,40 @@ export default class FleetingNotesPlugin extends Plugin {
...
+			string,
+			ObsidianNote
+		>();
+		var existingNotes = await this.getExistingFleetingNotes(folder);
+		existingNotes.forEach((note) =>
+			existingNotePathMap.set(note.file.path, note)
+		);
+		const hasSourceInMetaData = (note: ObsidianNote) => {
+			let hasSource = false;
+			note.frontmatter
```
Comment (matthewwong525) [inline src/main.ts:445]: this is just my opinion but an `if` statement might be easier to understand in this scenario

File: src/main.ts [hunk 4]
```diff
@@ -112,6 +119,38 @@ export default class FleetingNotesPlugin extends Plugin {
...
 	}
 
+	async insertSameSourceNotes() {
+		try {
+			let source: string;
+			new InputModal(this.app, "Enter Source", "Source", (result) => {
+				source = result;
+			}).open();
+
+			//i feel like this is a bad way to do this lmao
```
Comment (matthewwong525) [inline src/main.ts:129]: The below logic from lines 130 - 143 should be within the `onSubmit` function in line 126. This is kind of sketch lol

File: src/main.ts [hunk 5]
```diff
@@ -112,6 +119,38 @@ export default class FleetingNotesPlugin extends Plugin {
...
+			}
+			const sameSourceNotes = await this.getNotesWithSameSource(
+				this.settings.fleeting_notes_folder,
+				source
+			);
+			const template = "![[${linkText}]]\n\n";
+			const sameSourceNoteString = this.unprocessedNotesToString(
+				sameSourceNotes,
+				this.app.workspace.getActiveFile().path,
+				template
```
Comment (matthewwong525) [inline src/main.ts:141]: i dont see template in the function params of unprocessedNotesToString :O

File: src/main.ts (hunk 1 above)
Comment (ransurf) [reply src/main.ts:367]: I initially combined them but there was something different I noticed between them, I'll take a look again with a non-tired brain

**Pull request merged**